src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/AbstractTypesShouldNotHaveConstructors.cs

```diff
@@ -43,7 +43,7 @@ public override void Initialize(AnalysisContext context)
         private static void AnalyzeSymbol(SymbolAnalysisContext context)
         {
             var symbol = context.Symbol as INamedTypeSymbol;
-            if (symbol.IsAbstract)
+            if (symbol.IsAbstract && symbol.GetResultantVisibility() == SymbolVisibility.Public)
```

Comment:
We're doing this often, worth a utility extension `IsPublic()`?

Reply:
Done. I used the name `IsExternallyVisible` to match FxCop terminology.